source/common/http/context_impl.cc

```diff
@@ -4,7 +4,8 @@ namespace Envoy {
 namespace Http {
 
 ContextImpl::ContextImpl(Stats::SymbolTable& symbol_table)
-    : code_stats_(symbol_table), user_agent_context_(symbol_table) {}
+    : code_stats_(symbol_table), user_agent_context_(symbol_table),
+      async_client_stat_prefix_(user_agent_context_.pool_.add("http.async-client")) {}
```

Comment:
I notice this doesn't have the trailing '.' that it's replacing. I assume that is OK given the underlying impl but do we have test that actually look at these stats to make sure we didn't break them? We tend to have regressions when we make these types of changes.

Reply:
Yes, because the way prefixes get joined symbolically as a separate token, which get serialized into strings with intervening `"."`. There are a bunch of unit-tests around this functionality generally, and I also found there is an integration-test covering this specific case in https://github.com/envoyproxy/envoy/blob/ebdcf7cc28dfeb555a957096231fdec343de4709/test/integration/xfcc_integration_test.cc#L639

I have verified that fails if I change a character in the string.

Reply:
I lied -- tests all pass when I bork the name. Will add one.

Reply:
ok added a new test to cover that this prefix gets applied correctly.